src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,154 @@
+package duke;
+
+import duke.exceptions.DukeException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.nio.file.Path;
+
+/**
+ * Parser to parse user commands.
+ */
+public class Parser {
+    UI ui;
+    TaskArrayList taskList;
+    Path storagePath;
+
+    /**
+     * Constructor for Parser.
+     *
+     * @param ui UI object to print to.
+     * @param taskList TaskArrayList object to work with.
+     * @param storagePath Path object representing path to savefile.
+     */
+    Parser(UI ui, TaskArrayList taskList, Path storagePath) {
+        this.ui = ui;
+        this.taskList = taskList;
+        this.storagePath = storagePath;
+    }
+
+    /**
+     * Runs a user input string.
+     *
+     * @param userInput String keyed in by user.
+     * @return isExit, true if duke should exit after this command.
+     * @throws DukeException if unable to run argument.
+     */
+    public boolean run(String userInput) throws DukeException {
+        String[] cmd_args = userInput.split(" ", 2);
+        Task toAdd;
+
+        // switch case to split by command
+        switch (cmd_args[0]) {
+        case ("bye"):
+            this.runBye(cmd_args);
+            return true;
+
+        case ("list"):
+            runList(cmd_args);
+            break;
+
+        case ("done"):
+            runDone(cmd_args);
+            break;
+
+        case ("delete"):
+            runDelete(cmd_args);
+            break;
+
+        case ("find"):
+            runFind(cmd_args);
+            break;
+
+        case ("todo"):
+            runTodo(cmd_args);
+            break;
+
+        case ("deadline"):
+            runDeadline(cmd_args);
+            break;
+
+        case ("event"):
+            runEvent(cmd_args);
+            break;
+
+        default:
+            throw new DukeException("Unrecognised command");
+        }
```

Comment:
I like this too! Simple and easy to read.

Reply:
Agree with atyhamos, and I like the spaces between each case block, it is very helpful 👍